moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -212,27 +212,29 @@ bool CurrentStateMonitor::haveCompleteStateHelper(const ros::Time& oldest_allowe
   return (missing_joints == nullptr) || missing_joints->empty();
 }
 
-bool CurrentStateMonitor::waitForCurrentState(const ros::Time t, double wait_time) const
+bool CurrentStateMonitor::waitForCurrentState(const std::vector<const moveit::core::JointModel*>& joint_model_group,
+                                              const ros::Time t, double wait_time) const
 {
   ros::WallTime start = ros::WallTime::now();
   ros::WallDuration elapsed(0, 0);
   ros::WallDuration timeout(wait_time);
 
-  boost::mutex::scoped_lock lock(state_update_lock_);
-  while (current_state_time_ < t)
+  while (elapsed < timeout)
   {
-    state_update_condition_.wait_for(lock, boost::chrono::nanoseconds((timeout - elapsed).toNSec()));
-    elapsed = ros::WallTime::now() - start;
-    if (elapsed > timeout)
     {
-      ROS_INFO_STREAM_NAMED(LOGNAME,
-                            "Didn't receive robot state (joint angles) with recent timestamp within "
-                                << wait_time << " seconds.\n"
-                                << "Check clock synchronization if your are running ROS across multiple machines!");
-      return false;
+      boost::mutex::scoped_lock lock(state_update_lock_);
```

Comment:
I think you should remove those extra brackets since they limit the scope of the lock to the condition variable but there is also the comparison in line 230 that is not thread-safe